### PR TITLE
refactor(http): remove all facade methods from http module

### PR DESCRIPTION
**Pull request by alfaproject**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/@angular/http/src/backends/browser_jsonp.ts [hunk 1]
```diff
@@ -7,15 +7,17 @@
  */
 
 import {Injectable} from '@angular/core';
-import {global} from '../facade/lang';
+
+// Get the global object (https://github.com/tc39/proposal-global)
+const global: {[key: string]: any} = typeof this === 'object' ? this : Function('return this')();
```
Comment (alfaproject) [inline modules/@angular/http/src/backends/browser_jsonp.ts]: Can we import the TC39 shim instead?
https://www.npmjs.com/package/system.global

File: modules/@angular/http/src/body.ts [hunk 2]
```diff
@@ -51,7 +51,7 @@ export abstract class Body {
       return '';
     }
 
-    if (isJsObject(this._body)) {
+    if (typeof this._body === 'object') {
```
Comment (DzmitryShylovich) [inline modules/@angular/http/src/body.ts]: `if (typeof this._body === 'object' && this._body !== null ) {`

File: modules/@angular/http/src/http.ts [hunk 3]
```diff
@@ -35,11 +34,8 @@ function mergeOptions(
       responseType: providedOpts.responseType
     }));
   }
-  if (isPresent(method)) {
-    return newOptions.merge(new RequestOptions({method: method, url: url}));
-  } else {
-    return newOptions.merge(new RequestOptions({url: url}));
-  }
+
+  return newOptions.merge(new RequestOptions({method: method, url: url}));
```
Comment (vicb) [inline modules/@angular/http/src/http.ts]: nit: `return newOptions.merge(new RequestOptions({method, url}));`

File: modules/@angular/http/test/backends/jsonp_backend_spec.ts [hunk 4]
```diff
@@ -14,26 +14,21 @@ import {JSONPBackend, JSONPBackend_, JSONPConnection, JSONPConnection_} from '..
...
 
   addEventListener(type: string, cb: (data: any) => any) { this.callbacks.set(type, cb); }
 
   removeEventListener(type: string, cb: Function) { this.callbacks.delete(type); }
 
-  dispatchEvent(type: string, argument?: any) {
-    if (!isPresent(argument)) {
-      argument = {};
-    }
+  dispatchEvent(type: string, argument: any = {}) {
```
Comment (vicb) [inline modules/@angular/http/test/backends/jsonp_backend_spec.ts]: wouldn't this change the public API ?

`cb(argument || {});`

Comment (alfaproject) [inline modules/@angular/http/test/backends/jsonp_backend_spec.ts]: This is a spec, so should be good

Comment (vicb) [inline modules/@angular/http/test/backends/jsonp_backend_spec.ts]: yep, nv

**Comment by vicb** [comment]
Thanks for the PR !

**Pull request merged**